Bigtable/src/Connection/Grpc.php

```diff
@@ -371,47 +378,270 @@ public function readRow(array $args)
      */
     public function readRows(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $tableName = $this->pluck('tableName', $args);
+        $optionalArgs = [];
+        if (isset($args['options'])) {
+            $options = $this->pluck('options', $args);
+            if (isset($options['appProfileId'])) {
+                $optionalArgs['appProfileId'] = $this->pluck('appProfileId', $options);
+            }
+            if (isset($options['rows'])) {
+                $optionalArgs['rows'] = $this->rowSetObject($this->pluck('rows', $options));
+            }
+            if (isset($options['filter'])) {
+                $optionalArgs['filter'] = $this->rowFilterObject($this->pluck('filter', $options));
+            }
+            if (isset($options['rowsLimit'])) {
+                $optionalArgs['rowsLimit'] = $this->pluck('rowsLimit', $options);
+            }
+        }
+        return $this->send([$this->bigtableClient, 'readRows'], [
+            $tableName,
+            $optionalArgs,
+            $this->addResourcePrefixHeader($args, $tableName)
+        ]);
+    }
+
+    /**
+     * @param array $args
+     * @return RowSet
+     */
+    private function rowSetObject(array $args)
+    {
+        $rowSet = [];
+        if (isset($args['rowkeys'])) {
+            $rowSet['row_keys'] = $this->pluck('rowkeys', $args);
```

Comment:
Why are we using snake case here? Can't we maintain camelcase and let the serializer handle the conversion?